### PR TITLE
Add support for LVM to filesystem module

### PR DESCRIPTION
**Pull request by mscherer**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
filesystem

##### SUMMARY
While using LVM2_member is not very userfriendly, I do have another patch for that. But I am not sure where to mention the list of supported FS, should it go in the description of fstype ?

### CODE REVIEW

**Comment by bcoca** [comment]
sounds like a good place

**Comment by mattclay** [comment]
CI failure due to module docs validation error:

```
2017-02-27 15:32:05 ============================================================================
2017-02-27 15:32:05 lib/ansible/modules/system/filesystem.py
2017-02-27 15:32:05 ============================================================================
2017-02-27 15:32:05 ERROR:305:DOCUMENTATION.options.fstype.description.1: expected basestring @ data['options']['fstype']['description'][1]. Got None
```

**Comment by mscherer** [comment]
Repushed since it wasn't merged for 2.3

File: lib/ansible/modules/system/filesystem.py
Comment (pilou-) [inline lib/ansible/modules/system/filesystem.py:183]: `_get_fs_size` method need to be modified in order to handle `pvdisplay` tools.

It would be better to use a method instead of a string for the `fsinfo` value.

**Review by pilou- — CHANGES_REQUESTED** [review]

**Comment by mscherer** [comment]
@pilou- I did address the comment, but I am not gonna rewrite the whole module to have a object based design yet (even if I think that might be cleaner)

File: lib/ansible/modules/system/filesystem.py
Comment (pilou-) [inline lib/ansible/modules/system/filesystem.py]: `block_count` is a string, should not `int(size[:-1])` be used here ?

File: lib/ansible/modules/system/filesystem.py
Comment (pilou-) [inline lib/ansible/modules/system/filesystem.py]: You should add `choices: [ext4, ext4dev, ext3, ext2, xfs, btrfs, reiserfs, lvm]` and remove this line: [description should not list the options values ](http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-block).

**Comment by mscherer** [comment]
Repushed again since it was tagged stale_ci

**Comment by pilou-** [comment]
shipit

@mscherer :

1. please in order to get rid of the [`needs_revision`](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#label-needs_revision) label, please add a comment containing [`ready_for_review`](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#cmd-ready_for_review)
2. you're a maintainer of another module in the same namespace, then Ansibot adds automatically one `shipit` (Ansibot considers you have approved this pull request).
3. then in order to get this pull request auto merged, you need
    - either one approval (that is a comment with `shipit`) from a module (or namespace) maintainer
    - or three (in fact two, I already commented with `shipit`) approvals from anyone else.

@EvanK @LinusU (as randomly selected maintainer of `system` namespace) could you please review this pull request ? At least one `shipit` comment (from maintainer or namespace maintainer or core team member) is required in order to [automerge](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#community) this pull request.

**Comment by LinusU** [comment]
Nice 👍 

shipit

**Comment by mscherer** [comment]
ready_for_review

**Review by pilou- — COMMENTED** [review]
Look good to me

**Review by pilou- — APPROVED** [review]

**Comment by pilou-** [comment]
@ansibot ping ?

**Comment by mscherer** [comment]
bot_status

File: lib/ansible/modules/system/filesystem.py
Comment (pilou-) [reply lib/ansible/modules/system/filesystem.py:183]: Done (`pvs` is used now).

File: lib/ansible/modules/system/filesystem.py
Comment (bcoca) [inline lib/ansible/modules/system/filesystem.py:26]: you set choices, but this is actually not restricted in args spec

File: lib/ansible/modules/system/filesystem.py
Comment (bcoca) [reply lib/ansible/modules/system/filesystem.py:183]: not something changed in this PR, but I think this would be better as default matching the named passed in (most fs follow same pattern) and only put in entries for what needs to be overriden

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/system/filesystem.py
Comment (pilou-) [reply lib/ansible/modules/system/filesystem.py:26]: Isn't the check made [L213](https://github.com/mscherer/ansible/blob/c19a95a315680a3f68b157054aa54c0ecc31ee28/lib/ansible/modules/system/filesystem.py#L214) sufficient ?

Comment (mscherer) [reply lib/ansible/modules/system/filesystem.py:26]: I guess I can combine the 2 by just reusing keys from the dict as arguments.

**Comment by mscherer** [comment]
@bcoca I think I did implement what was blocking in the last review, something else missing ?

**Comment by duck-rh** [comment]
@bcoca any reason why this could not fly? (this PR is old and seems to have reach approval so…)

**Review by bcoca — APPROVED** [review]

**Comment by bcoca** [comment]
rebiuld_merge

File: lib/ansible/modules/system/filesystem.py
Comment (bcoca) [inline lib/ansible/modules/system/filesystem.py:30]: 2.5 at this point but i'll fix after merge

Comment (mattclay) [reply lib/ansible/modules/system/filesystem.py:30]: @bcoca You could push another commit to this PR to fix that before merging.

Comment (mattclay) [reply lib/ansible/modules/system/filesystem.py:30]: It needs to be retested by CI anyways since the CI result is stale.

Comment (bcoca) [reply lib/ansible/modules/system/filesystem.py:30]: @mattclay rebuild_merge is supposed to make the bot rerun CI and only merge if it succeeds

**Comment by duck-rh** [comment]
@bcoca thanks for quickly responding. Btw you made a typo in your command, so not sure the bot is able to deal with this :-)

**Comment by bcoca** [comment]
@duck-rh no, its my super secret command for the bot to autocorrect ....

rebuild_merge

**Pull request merged**
